test/vcf/debugulator_integration_test.cpp

```diff
@@ -52,79 +52,124 @@ namespace ebi
-          CHECK_FALSE(is_valid_vcf_file(file, validator, reports));
-
-
-          // now we fix the file
-          // file: rewind the same ifstream as before
-          // report: SqliteReportRW to read the errors. it must be flushed before this
-          // fixed_vcf: writing result vcf to a stringstream in memory
-          file.clear();
-          file.seekg(0);
-          report->flush();
-          std::stringstream fixed_vcf;
-          vcf::debugulator::fix_vcf_file(file, *report, fixed_vcf);
-          file.close();
-
-          // here we check that the fixed file is valid:
-          // fixed_vcf: stringstream the debugulator::fix_vcf_file wrote
-          // reports: ReportWriters (one SqliteReportRW writing to database "errors_after")
-          fixed_vcf.seekg(0);
-          auto validator2 = ebi::vcf::FullValidator_v41{
-                  std::make_shared<ebi::vcf::Source>(source),
-                  std::make_shared<std::vector<ebi::vcf::Record>>(records)};
-          auto db_path_after = boost::filesystem::path{"/tmp/"} / path.filename();
-          db_path_after += ".debugulator_test.errors_after.db";
-          boost::filesystem::remove(db_path_after);   // make sure the dbs doesn't exist from previous runs
-          auto output_after = new ebi::vcf::SqliteReportRW{db_path_after.string()};
-          reports.clear();
-          reports.emplace_back(output_after);
-          bool successfully_fixed = is_valid_vcf_file(fixed_vcf, validator2, reports);
-
-          CHECK(successfully_fixed);
-
-          auto fixed_vcf_path = boost::filesystem::path{"/tmp/"} / path.filename();
-          fixed_vcf_path += ".debugulator_test.fixed.vcf";
-          boost::filesystem::remove(fixed_vcf_path);
-          if (not successfully_fixed) {
-              // dump fixed file to manually inspect
-              std::ofstream fixed_file{fixed_vcf_path.string()};
-              fixed_file << fixed_vcf.str();
-              fixed_file.close();
-              std::cout << "This test failed, you can see the related data at: " << db_path_after.string() << ", ";
-              std::cout << db_path.string() << " and " << fixed_vcf_path.string() << std::endl;
-          } else {
-              // clean report dbs
-              boost::filesystem::remove(db_path);
-              boost::filesystem::remove(db_path_after);
+          // semantic error, the number of values doesn't match the description in the meta section
+          path = boost::filesystem::path("test/input_files/v4." + std::to_string(i) + "/failed/failed_body_info_034.vcf");
+          SECTION(path.string())
+          {
+              std::tie(first_validation, validation_after_fixing, debug_message, lines) = fix_file(path, version);
+              CHECK_FALSE(first_validation);
+              INFO(debug_message);
+              CHECK(validation_after_fixing);
+              CHECK(lines == 5);
           }
+          version = static_cast<ebi::vcf::Version>(i);
       }
   }
 }
-
```

Comment:
It doesn't sound very useful to create a copy of the VCF when no errors are present, but it would be interesting to make the debugulator stop (more than fail) when the report is empty.